pkg/operator/backupinstance/operator.go

```diff
@@ -0,0 +1,97 @@
+package backupinstance
+
+import (
+	crv1 "github.com/grtl/mysql-operator/pkg/apis/cr/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/grtl/mysql-operator/pkg/client/clientset/versioned"
+	"github.com/grtl/mysql-operator/pkg/logging"
+	"github.com/grtl/mysql-operator/pkg/util"
+)
+
+const (
+	jobCreateTemplate = "artifacts/backupinstance-job-create.yaml"
+	jobDeleteTemplate = "artifacts/backupinstance-job-delete.yaml"
+)
+
+// Operator represents an object to manipulate Backup custom resources.
+type Operator interface {
+	CreateBackup(backup *crv1.MySQLBackupInstance) error
+	DeleteBackup(backup *crv1.MySQLBackupInstance) error
+}
+
+// NewOperator returns a new Operator.
+func NewOperator(clientset versioned.Interface, kubeClientset kubernetes.Interface) Operator {
```

Comment:
By precedent set by the cluster operator, this should be named `NewBackupScheduleOperator`.